ext/mysqli/mysqli_nonapi.c

```diff
@@ -359,11 +359,8 @@ void mysqli_common_connect(INTERNAL_FUNCTION_PARAMETERS, bool is_real_connect, b
 	} else {
 		(Z_MYSQLI_P(object))->ptr = mysqli_resource;
 	}
-	if (!is_real_connect) {
-		return;
-	} else {
-		RETURN_TRUE;
-	}
+
+	RETURN_TRUE;
```

Comment:
I don't think we can change this. At least not like that. Constructors can't return a type

Reply:
You are right, I'll fix this.

Reply:
It's supposed to be fixed with the last commit

Reply:
Forgive me, but I don't see how the last commit fixes it. You need a special case for `if(!is_real_connect)`. This method can't return true or false always. You also need an unconditional return to prevent fallthrough. So, in other words, I don't think we can modify this piece of code at all. Either way, one of the 5 variants will be affected.

Reply:
I believe what you need to do is move the existing code into the !object branch and have an unconditional RETURN_TRUE in the other branch.

Return value for ctor doesn't matter, it will just get ignored (see existing RETURN_FALSEs).

Reply:
> Forgive me, but I don't see how the last commit fixes it.

I completely missed that `MYSQLI_RETURN_RESOURCE` won't actually return anything. :( I'd happy if the macro could be renamed to something like `MYSQLI_RETVAL_RESOURCE()` to avoid further confusion.

> I believe what you need to do is move the existing code into the !object branch and have an unconditional RETURN_TRUE in the other branch.

Thank you for the suggestion! I've just pushed it (by also reordering the if branches for easier understanding)

Reply:
> > Forgive me, but I don't see how the last commit fixes it.
> 
> I completely missed that `MYSQLI_RETURN_RESOURCE` won't actually return anything. :( I'd happy if the macro could be renamed to something like `MYSQLI_RETVAL_RESOURCE()` to avoid further confusion.

Yeah, let's do that.

Reply:
Added a new commit for that as well as rebase to master for the funcinfo change